purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -2074,6 +2075,9 @@ class Purchases internal constructor(
         fun configure(
             configuration: PurchasesConfiguration
         ): Purchases {
+            if (isConfigured) {
```

Comment:
This is not thread safe, so if trying to configure the SDK multiple times in different threads, there is a chance of missing this. We should refactor the `Purchases` instance a bit to make sure the shared instance is thread-safe, same way we have in iOS. Created a [followup ticket](https://linear.app/revenuecat/issue/SDK-2970/make-purchases-instance-creation-thread-safe-[android]) to handle this

Reply:
👍🏻 
iOS used to have the same issue. IIRC I made it thread safe to fix race conditions in tests.